### ISSUE TITLE
Integrate with fzf?

### ISSUE THREAD

**Issue by cheecheeo** [ctrlpvim/ctrlp.vim#111]
FZF is insanely fast and more general purpose it would be _awesome_ if ctrlp would detect if it was installed and use it https://github.com/junegunn/fzf

**Comment by mattn** [comment]
FZF is UI as same as ctrip. so I cant understand you expected.

On 8/8/15, John Chee <notifications@github.com> wrote:
> FZF is insanely fast and more general purpose it would be _awesome_ if ctrlp
> would detect if it was installed and use it https://github.com/junegunn/fzf
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/ctrlpvim/ctrlp.vim/issues/111


-- 
- Yasuhiro Matsumoto

**Comment by ggVGc** [comment]
Yeah, fzf and ctrlp are mutually exclusive imo. Personally I am slowly transitioning towards fzf over ctrlp.

**Comment by cheecheeo** [comment]
@mattn Yes, I agree they have similar UIs. The feature request here is that ctrlp could detect if fzf is already installed on the machine and then if so, use fzf rather than it's own fuzzy-finder-in-vimscript implementation. ctrlp has better vim integration out-of-the-box than fzf, so I think ctrlp still wins there, but for performance with a large number of files fzf wins hands down.

So to reiterate request here is for ctrlp to keep on providing great vim integration but if fzf is installed to delegate it's fuzzy finding functionality to fzf. If fzf is not installed it can continue to rely on the fuzzy-finder-in-vimscript implementation.

**Comment by Mudox** [comment]
For a faster fuzzy match engine, I think [ctrlp-cmatcher](https://github.com/JazzCore/ctrlp-cmatcher) extension is worth a try.
AFAIK, fzf is only available on terminals, not on GUIs.

**Comment by mattn** [comment]
I use https://github.com/nixprime/cpsm

**Comment by ming-codes** [comment]
fzf have a non-interactive mode you can use with option `-f`. You can integrate with ctrlp with `let g:ctrlp_user_command` var.

I haven't have much success with this yet. I'll report back when I have more to share.

**Comment by ming-codes** [comment]
Whoa...deja vu!

So, I've been using fzf for 2 years now. It's blazingly fast, but I do miss the caching and mru of ctrlp and wanting the switch back. Upon switching back to ctrlp, the speed is really making me to have second thought.

So yea, ctrlp has the feature I want, but not the speed I want. So here I am searching the ctrlp issues list for solution and stumbled upon my own previous comment! Ha!

**Comment by mattn** [comment]
wondering which part do you say slow.

* listing files
* filtering on prompt

**Comment by HaleTom** [comment]
It would be great to be able to specify a path-like list of executables:

`fzy:fzf:peco:percol:selecta`

And perhaps listers:

`fd -options:find -options`

**Comment by mattn** [comment]
If you expected integration with native UI instead of CtrlP, you've better to use fzf or vim-fz.

**Comment by HaleTom** [comment]
@mattn I was hoping for integration with CtrlP as it's by far the most feature-complete.
